Rename TempDirectoryBuilder::create

IohannRabeson opened the pull request.

IohannRabeson committed:
Support for content copied from a file.

IohannRabeson committed:
Removed the support for YAML.

IohannRabeson committed:
Drop by default.

IohannRabeson committed:
Ensure the random temp directory does not already exists.

IohannRabeson committed:
Fix ci.yml

IohannRabeson committed:
Merge pull request #1 from IohannRabeson/copy_file

Add support for file content and cleanup.

IohannRabeson committed:
Fix format.

IohannRabeson committed:
Cleanup doc and metadata.

IohannRabeson committed:
Fix CI (#2)

* Fix CI by cleaning up the doc and format

IohannRabeson committed:
Merge branch 'kaplanelad:main' into main

IohannRabeson committed:
Reboot (#3)

Rewrite a huge part of the library and rename it to temp-dir-builder.

IohannRabeson committed:
Allow to create binary files.

IohannRabeson committed:
Update README.md

IohannRabeson committed:
Documentation.

IohannRabeson committed:
Produce an error if an entry is declared outside the directory.

IohannRabeson committed:
Add a dedicated error for the case where the entry name is empty.

IohannRabeson committed:
Update README.md

IohannRabeson committed:
Fix CI and reduce possibility of panics.

IohannRabeson committed:
Renaming

kaplanelad commented:
Hey @IohannRabeson,

Would you like to share what you're trying to solve? I'd love to see if it's something I can incorporate into this crate.

IohannRabeson commented:
Hi @kaplanelad!
Yes, in my own fork I added:
 - the ability to create file by copying an existing file
 - the ability to create a file by specifying its binary content
 - the ability to create directories
 - refactored how the errors are reported (by implementing a proper error struct)
These 2 features was problematic with the YAML definition system, and I did not need it so I removed that in my own fork. This allowed me to do a huge cleanup (a lot of stuff was public for no reason). Also you were making public the internal of the library which was a design issue.
When I decided to remove the YAML support I was pretty sure you will be against this change so I did my own fork.

kaplanelad commented:
you are more then welcome to open issues and we can implement it together on my crate instead of forking

IohannRabeson commented:
Sorry, it was much faster and easier to fork instead of reporting each issues, discussing them and fixing them (too much issues and architectures design flaws - particularly the fact that you expose the internals of the library e.g FileMetadata being public and being used for deserialization).
I also took the opportunity to rename everything and even the library itself as "tree_fs" does not reflect what the library is doing and I also wrote a test suite that cover more code.
I think I rewrote everything 😄 

What really prevented me to propose changes was the YAML support, to me this is a false good idea. You add an another layer of complexity where you want instead something that is simple and reliable. By adding support for YAML you add all the issues related to parsing. In my fork, you **must** define your temp directory tree structure as **plain rust code**, so the compiler will produces errors **at compile time** if there is a syntax error. With YAML (or any of format) you will get errors at runtime, when running the tests and this is definitely not acceptable to me.

kaplanelad commented:
ok, good luck
